creature.cpp

```diff
@@ -1144,6 +1144,8 @@ static string getAttackParam(AttackType type) {
     case AttackType::SPELL: return "spell";
     case AttackType::POSSESS: return "touch";
   }
+  assert("__FUNCTION__ case unknown attack type");
+  return "";
```

Comment:
True, tbh I forgot C++ did that with case and enums. I just had to go double check the gcc man.  I was mainly focused on palliatively cleaning out the compiler warnings without turning too many off as in my experience, that's always a bad idea.

What version of clang are you using as it doesn't compile on 3.5 here.

Just curious, why the custom FAIL instead of the assert?